packages/flutter_tools/templates/list_detail_app/lib/src/dummy_feature/dummy_item_list_view.dart.tmpl

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+
+import '../settings/settings_view.dart';
+import 'dummy_item.dart';
+import 'dummy_item_details_view.dart';
+
+/// Displays a list of DummyItems.
+class DummyItemListView extends StatelessWidget {
+  const DummyItemListView({
+    Key key,
+    this.items = const [DummyItem(1), DummyItem(2), DummyItem(3)],
+  }) : super(key: key);
+
+  static const routeName = '/';
+
+  final List<DummyItem> items;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Dummy Items'),
+        actions: [
+          IconButton(
+            icon: const Icon(Icons.settings),
+            onPressed: () {
+              // Navigate to the settings page. If the user leaves and returns
+              // to the app, the navigation stack is restored.
+              Navigator.restorablePushNamed(context, SettingsView.routeName);
+            },
+          ),
+        ],
+      ),
+
+      // To work with lists that may contain a large number of items, it’s best
+      // to use the ListView.builder constructor.
+      //
+      // In contrast to the default ListView constructor, which requires
+      // building all Widgets up front, the ListView.builder constructor lazily
+      // builds Widgets as they’re scrolled into view.
+      body: ListView.builder(
```

Comment:
The listview also needs a restoration id for full state restoration of scroll position

Reply:
Thanks, added the following to the `ListView.builder`:

```dart
        // Providing a restorationId allows the ListView to restore the
        // scroll position when a user leaves and returns to the app after it
        // has been killed while running in the background.
        restorationId: 'dummyItemListView',
```